agent/src/beerocks/slave/backhaul_manager/internal/expected_ap_metrics_response.h

```diff
@@ -0,0 +1,97 @@
+/* SPDX-License-Identifier: BSD-2-Clause-Patent
+ *
+ * SPDX-FileCopyrightText: 2016-2020 the prplMesh contributors (see AUTHORS.md)
+ *
+ * This code is subject to the terms of the BSD+Patent license.
+ * See LICENSE file for more details.
+ */
+
+#ifndef _EXPECTED_AP_METRICS_RESPONSE_H
+#define _EXPECTED_AP_METRICS_RESPONSE_H
+
+#include <bcl/beerocks_defines.h>
+#include <tlvf/CmduMessageTx.h>
+#include <tlvf/common/sMacAddr.h>
+#include <tlvf/tlvftypes.h>
+#include <unordered_set>
+
+namespace beerocks {
+
+class ExpectedApMetricsResponse {
+public:
+    /**
+     * @brief resets this object with a new mid
+     * @details resets all members of this object to its empty initial state except for the mid 
+     * that is set as was given bt the parameter (mid default value is zero (0))
+     * @param mid the new mid value
+     * @return radio mac of the found slave if found, otherwise empty string
+     */
+    void reset_to_new_mid(uint16_t mid = 0);
+
+    /**
+     * @brief returns the mid of this object
+     * @return mid
+     */
+    uint16_t get_mid() const;
+
+    /**
+     * @brief creates and returns a AP-RESPONSE-METRICS Tx message 
+     * @return a fresh new CmduMessage with the correct type and mid
+     */
+    ieee1905_1::CmduMessageTx &create_tx_message();
+
+    // bssid related operations
+
+    /**
+     * @brief stores expected bssids
+     * @details adds the bssids in the range [first,last) to the already existing internal list
+     * @param first iterator to the source begin
+     * @param last iterator to the source last
+     * @return - nothing
+     */
+    template <class FirstIt, class LastIt> void add_expected_bssid(FirstIt first, LastIt last);
```

Comment:
Why do you use template here? There is only one use if this function, so please replace it to non-template function.

Reply:
Template is good, because different containers have a different iterator type.

However, first and last must be the same type, so
```suggestion
    template <class Iterator> void add_expected_bssid(Iterator first, Iterator last);
```

Reply:
But here it is redundant because we don't have different containers. We have a single container.